lib/errors.ts

```diff
@@ -126,56 +126,81 @@ const messages: {
 	`,
 
 	BalenaExpiredToken: () => stripIndent`
-		Looks like your session token is expired.
+		Looks like your session token has expired.
 		Please try logging in again with:
 			$ balena login`,
 };
 
+const EXPECTED_ERROR_REGEXES = [
+	/^BalenaApplicationNotFound:/, // balena-sdk
+	/^BalenaDeviceNotFound:/, // balena-sdk
+	/^Missing \w+$/, // Capitano, oclif parser: RequiredArgsError, RequiredFlagError
+	/^Unexpected arguments?:/, // oclif parser: UnexpectedArgsError
+	/to be one of/, // oclif parser: FlagInvalidOptionError, ArgInvalidOptionError
+];
+
 export async function handleError(error: any) {
+	// Set appropriate exitCode
 	process.exitCode =
 		error.exitCode === 0
 			? 0
 			: parseInt(error.exitCode, 10) || process.exitCode || 1;
 
+	// Handle non-Error objects (probably strings)
 	if (!(error instanceof Error)) {
 		printErrorMessage(String(error));
 		return;
 	}
 
+	// Prepare message
 	const message = [interpret(error)];
 
-	if (process.env.DEBUG && error.stack) {
-		message.push('\n' + error.stack);
-	}
-	printErrorMessage(message.join('\n'));
-
-	const expectedErrorREs = [
-		/^BalenaApplicationNotFound:/, // balena-sdk
-		/^BalenaDeviceNotFound:/, // balena-sdk
-		/^Missing \w+$/, // Capitano's command line parsing error
-		/^Unexpected arguments?:/, // oclif's command line parsing error
-	];
-
-	if (
-		error instanceof ExpectedError ||
-		expectedErrorREs.some(re => re.test(message[0]))
-	) {
-		return;
+	if (error.stack) {
+		if (process.env.DEBUG) {
+			message.push('\n' + error.stack);
+		} else {
+			// Include first line of stacktrace
+			message.push('\n' + error.stack.split(`\n`)[0]);
+		}
 	}
 
-	// Report "unexpected" errors via Sentry.io
-	const Sentry = await import('@sentry/node');
-	Sentry.captureException(error);
-	await Sentry.close(1000);
-	// Unhandled/unexpected error: ensure that the process terminates.
-	// The exit error code was set above through `process.exitCode`.
-	process.exit();
+	// Expected?
+	const isExpectedError =
+		instanceOf(error, ExpectedError) ||
```

Comment:
Using the custom instanceOf function here, as intanceof may fail.

Reply:
Harmless, but the custom `instanceOf` function is only useful when the class (`ExpectedError`) is defined in an external package/repo (other than `balena-cli`), because of the risk of there being different installed versions of the external package in different `node_modules` folders (each such version listed in the `npm-shrinkwrap.json` file). In this case, `ExpectedError` is defined in `balena-cli` itself, in fact defined in this same file, so no such risk.

Reply:
Ah yeah, will revert that.